src/melee.cpp

```diff
@@ -1940,3 +1945,129 @@ double player::unarmed_value() const
     // TODO: Martial arts
     return melee_value( ret_null );
 }
+
+void player::disarm( npc &target )
+{
+    if( !target.is_armed() ) {
+        return;
+    }
+
+    item &it = target.weapon;
+
+    const char *target_name = target.name.c_str();
+    const char *item_name = it.tname().c_str();
```

Comment:
I am not that good at this. How should I modify it? Assign a copy of the `tname()` to the local var?